### PR TITLE
[12.0][MIG] base_vat_optional_vies

### PR DESCRIPTION
**Pull request by Tardo**
Based on https://github.com/OCA/account-financial-tools/pull/750

cc @Tecnativa

### CODE REVIEW

**Comment by Tardo** [comment]
@sergio-teruel and @chienandalu please, review this PR.

**Review by chienandalu — APPROVED** [review]
Tested on runbot :+1: 

My only doubt is if there could be a more proper place to put the field. The partner form is a model that tends to be heavily extended and VIES info is not that relevant after all.

File: base_vat_optional_vies/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+# Copyright 2017 Tecnativa - David Vidal
+# Copyright 2019 FactorLibre - Rodrigo Bonilla
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+{
+    'name': "Optional validation of VAT via VIES",
+    'category': 'Accounting',
+    'version': '12.0.1.0.0',
+    'depends': [
+        'base_vat',
+        'base',
```
Comment (sergio-teruel) [inline base_vat_optional_vies/__manifest__.py]: You can remove base dependency

**Review by rafaelbn — APPROVED** [review]
Tested in runbot :+1:  no code review

@Tardo IMHO change "VIES validation passed" by "VIES validation" , passed is not necessary as the True value of the Boolean is understandable

@chienandalu the VIES validation is very relevant to know which kind of taxed you could apply in the European Union. If VIES validation is true you can use Fiscal Position "Intracomunicaria" so taxed will be change for this customers.

File: base_vat_optional_vies/models/res_partner.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        if partner:
+            # If there's an exception checking VIES, the upstream method will
+            # call simple_vat_check and thus the flag will be removed
+            partner.update({'vies_passed': True})
+        res = super(ResPartner, self).vies_vat_check(country_code, vat_number)
+        if not res:
+            return self.simple_vat_check(country_code, vat_number)
+        return res
+
+    @api.constrains('vat', 'commercial_partner_country_id')
```
Comment (sergio-teruel) [inline base_vat_optional_vies/models/res_partner.py]: In v12 the field commercial_partner_country_id not exists, so change by commercial_partner_id.country_id

File: base_vat_optional_vies/views/res_partner_view.xml [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+        <field name="name">Add VIES passed fields</field>
+        <field name="model">res.partner</field>
+        <field name="inherit_id" ref="base.view_partner_form"/>
+        <field name="arch" type="xml">
+            <field name="vat" position="after">
+                <field name="vies_passed"/>
+            </field>
+        </field>
+    </record>
+</odoo>
```
Comment (sergio-teruel) [inline base_vat_optional_vies/views/res_partner_view.xml]: Add blank line

**Review by sergio-teruel — CHANGES_REQUESTED** [review]

**Comment by Tardo** [comment]
Changes done, thanks!

File: base_vat_optional_vies/models/res_partner.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+        if partner:
+            # If there's an exception checking VIES, the upstream method will
+            # call simple_vat_check and thus the flag will be removed
+            partner.update({'vies_passed': True})
+        res = super(ResPartner, self).vies_vat_check(country_code, vat_number)
+        if not res:
+            return self.simple_vat_check(country_code, vat_number)
+        return res
+
+    @api.constrains('vat', 'commercial_partner.country_id')
```
Comment (MiquelRForgeFlow) [inline base_vat_optional_vies/models/res_partner.py:37]: Runbot is red. You can't do this. 
```yml
2019-07-19 12:51:10,107 161 WARNING openerp_test odoo.models: method res.partner.check_vat: @constrains parameter 'commercial_partner.country_id' is not a field name
```

**Review by MiquelRForgeFlow — COMMENTED** [review]
!!

File: base_vat_optional_vies/models/res_partner.py (hunk 4 above)
Comment (Tardo) [reply base_vat_optional_vies/models/res_partner.py:37]: Thanks! fixed in https://github.com/OCA/account-financial-tools/pull/868

**Pull request closed without merging**